x.payment_batching/payment_batching.md

```diff
@@ -0,0 +1,307 @@
+
+If we assume that consolidation transactions will pay only 20% of the
+feerate of a normal transaction and will consolidate 100 inputs at a
+time, we can calculate the savings of using the two-step procedure for
+our 10-to-1 and 1-to-1 scenarios above (while showing, for comparison,
+the simple best-case scenario of already having a large input available).
+
+![Saving rates for best, typical, and tough cases of payment batching after consolidation](img/p2wpkh-batching-after-consolidation.png)
+
+Most notably, the savings from consolidation allows the tough case go
+from worst performing to best performing.  For the typical case,
+consolidation actually loses money when only making a single payment,
+but when actually batching, it performs almost as well as the best case
+scenario.
+
+In addition to payment batching directly providing a fee savings,
+batching also uses limited block space more efficiently by reducing the
+number of vbytes per payment.  This increases the available supply of
+block space and so, given constant demand, can make it more affordable.
+In that way, increased use of payment batching may lower the feerate for
+all Bitcoin users.
+
+In summary, payment batching provides significant savings for services
+that typically have inputs available that are 5 to 20 times larger than
+their typical output.  For services not in that position, the savings
+from batching alone are smaller but perhaps still worth the effort;
+if the services are willing to also pre-consolidate their inputs, the
+savings can be quite dramatic.
+
+Note: the figures and plots above all assume use of P2WPKH inputs and
+outputs.  We expect that to become the dominant script type on the
+network in the future (until something better comes along).  However, if
+you use a different script type (P2PKH, or multisig using P2SH or
+P2WSH), the number of vbytes used to spend them are even larger, so the
+savings rate will be higher.
+
+## Problems
+
+The fee-reduction benefits of payment batching do create tradeoffs and
+problems that will need to be addressed by any service using the
+technique.
+
+### Delayed gratification
+
+This is the primary problem with payment batching.  Although some
+situations naturally lend themselves to payment batching (e.g.  sending
+out the company payroll for every employee at the same time), many
+services primarily send money to users when those users make a
+withdrawal request.  In order to batch payments, the service must get
+the user to accept that their payment will not be sent immediately---it
+will be held for some period of time and then combined with other
+withdrawal requests.
+
+One mitigation for the problem of delayed gratification is to allow the
+user to choose between an immediate payment and a delayed payment, with
+a different fee provided for each option.  For example:
+
+    [X] Free withdrawal (payment sent within 6 hours)
+    [ ] Immediate withdrawal (withdrawal fee 0.123 mBTC)
```

Comment:
Ideally we could use a made-up unit like bogocoin so that our documentation doesn't become outdated each time the price swings 10x either way.  But if we're going to use an actual denomination, I leave it to you and John (and Mike if he wants) to decide what to use.  As long as it isn't something ridiculous like tonal, I'll conform.